RELEASE/scripts/autoscend/auto_equipment.ash

```diff
@@ -716,13 +716,70 @@ void finalizeMaximize(boolean speculative)
 			addToMaximize("+equip " + toEquip);
 		}
 	}
+
+	if(in_wereprof() && auto_haveDarts()) //Absolutely need darts for Professor. Should level up darts while Werewolf too
+	{
+		if(is_werewolf())
+		{
+			addBonusToMaximize($item[Everfull Dart Holster], 1000);
+		}
+		else
+		{
+			addToMaximize("+equip " + $item[Everfull Dart Holster]);
+		}
+	}
+
+	if(in_wereprof() && !is_werewolf() && (possessEquipment($item[biphasic molecular oculus]) || possessEquipment($item[triphasic molecular oculus]))) //Want that Advanced Research as a professor
```

Comment:
would an improvement here be to only equip if there is a monster in the zone we are going to which we haven't researched already?

Reply:
Yes, it would. Investigating this, it seems like Advanced Research has never actually been cast in all of my runs by Autoscend! So, moving Advaced Research to Stage1 instead of Stage4 and testing from there. Once I start getting Advanced Research, I can take a look at whehter all of the monsters in the zone are in wereProfessorAdvancedResearch and if so, then we won't bother equipping the Oculus. I wonder if that will take into account URs, not that there are many zones affected by that?

Reply:
I found the issue. I believe my code is sound, but have_skill (Mafia function) returns false for Advanced Research despite the Oculus being equipped. So going to fix the stage 4 code to not check canUse and then figure out how to do the improvement that spurred this